pkg/jx/cmd/vault/vault.go

```diff
@@ -0,0 +1,25 @@
+package vault
+
+import (
+	"fmt"
+	"github.com/hashicorp/vault/api"
+)
+
+// GetSecrets returns a list of secrets from Vault
+// It's a bit hacky at the moment
+func GetSecrets(client *api.Client) ([]string, error) {
```

Comment:
I would design the interface a bit differently in order to hide`api.Client` underneath, and then you can generate a fake interface.

```Go
type Vaulter interface {
    Secrets() (map[string]string, error)
    Secret(name string) (string, error)
...
}

type JxVaulter struct {
     client *api.Client
}

func NewJxVaulter(client *api.Client) {
}

func (v *JxVaulter) Secrets() (map[string]string, error) {
}
```